packages/core/src/odata-common/uri-conversion/get-filter.ts

```diff
@@ -142,6 +144,18 @@ export function createGetFilter(uriConverter: UriConverter): GetFilter {
       ].join(' ');
     }
 
+    if (isBooleanFilterFunction(filter)) {
```

Comment:
This function is getting bigger and bigger. I think I started that and I know this goes a little beyond this PR. Still, maybe it is time to split it into smaller functions?

For example: 
```ts
...
if (isFilterLink(filter)) {
  return getODataExpressionForFilterLink(...);
}
if (isFilter(filter)) {
  return getODataExpressionForFilter(...);
}
...
```